### PR TITLE
Make anchors buttonable

### PR DESCRIPTION
**Pull request by alshakero**
Fixes: https://github.com/Starcounter/uniform.css/issues/186

Gentle reminder to review only `src` dir and `index.html`.

### CODE REVIEW

**Comment by alshakero** [comment]
@tomalec says:

> * is tricky, as it matches <template>, <imported-template>, and other undesired elements. I'd stick with YAGNI and button, a

And I agree. I'll address this.

**Comment by warpech** [comment]
@andwah Please provide a guideline when to use a flat button as a link that will be pasted to UniformDocs (https://uniform.starcounter.io/UniformDocs/Button page, I guess)

**Comment by alshakero** [comment]
@warpech this is to use a link as a button, not a button as a link.

**Comment by warpech** [comment]
I know, sorry for a mental shortcut. Thanks for pointing this out.

Actually we have a https://uniform.starcounter.io/UniformDocs/Url page, so that should definitely go there, with a cross link between the "Button" and "Link" pages.

File: dist/commando.unminified.css [hunk 1]
```diff
@@ -291,31 +292,70 @@ Buttons
...
 
 .uni-primary-button > ::slotted(button[disabled]),
 .uni-success-button > ::slotted(button[disabled]),
 .uni-danger-button > ::slotted(button[disabled]),
-.uni-flat-button > ::slotted(button[disabled]) {
+.uni-flat-button > ::slotted(button[disabled]),
+.uni-primary-button > ::slotted(a[disabled]),
+.uni-success-button > ::slotted(a[disabled]),
+.uni-danger-button > ::slotted(a[disabled]),
+.uni-flat-button > ::slotted(a[disabled]) {
```
Comment (warpech) [inline dist/commando.unminified.css]: Please remove lines that define rules for `disabled` attribute on `<a>`, beause it is against the HTML spec.

File: index.html [hunk 2]
```diff
@@ -783,6 +778,67 @@ <h2 id="button">Button</h2>
...
+                            </tr>
+                            <tr>
+                                <td>
+                                    <div class="uni-primary-button">
+                                        <a>Primary</a>
+                                    </div>
+                                </td>
+                                <td>
+                                    <div class="uni-primary-button">
+                                        <a disabled>Primary</a>
```
Comment (warpech) [inline index.html]: It is against the spec to have the `disabled` attribute here

File: index.html [hunk 3]
```diff
@@ -783,6 +778,67 @@ <h2 id="button">Button</h2>
                         </table>
                     </template>
                 </div>
+                
+                <p>Anchors will also be turned into a uniform button when wrapped with <code>uni-*-button</code></p>
```
Comment (warpech) [inline index.html]: "Uniform" should be capitalised.

**Comment by alshakero** [comment]
All addressed

File: src/uniform/uniform.css [hunk 4]
```diff
@@ -14,26 +14,27 @@ Buttons
 .uni-flat-button {
   display: inline-block;
 }
-.uni-flat-button {
-  min-width: auto;
```
Comment (warpech) [inline src/uniform/uniform.css]: What is the reason and consequence of this?

File: index.html [hunk 5]
```diff
@@ -783,6 +778,46 @@ <h2 id="button">Button</h2>
                         </table>
                     </template>
                 </div>
+                
+                <p>Anchors will also be turned into a Uniform button when wrapped with <code>uni-*-button</code></p>
+                <div data-shadow-host>
+                    <template is="declarative-shadow-dom" mode="open">
+                        <link rel="stylesheet" href="./dist/uniform.css">
+                        <table>
```
Comment (warpech) [inline index.html]: It doesn't make sense to have a table with a single column

File: src/uniform/uniform.css (hunk 4 above)
Comment (alshakero) [reply src/uniform/uniform.css]: I honestly don't remeember now why I removed this. I thought about adding it back but I don't see a reason why it should be there. 

1. It's not overriding anything.
2. Removing it does not change the demos at all.
3. It's hard to use the blame tool where this line come from. It is appearing in a big commit when I merged underwear and uniform.

File: index.html (hunk 5 above)
Comment (alshakero) [reply index.html]: Addressed.

File: index.html [hunk 6]
```diff
@@ -7,11 +7,6 @@
 
     <!-- Toggle it on and off -->
     <link rel="stylesheet" href="./dist/underwear.css">
-    <!--
-        We strongly suggest to use Uniform in shadow roots,
-        then the outer tree could be free of opinionated classes
-    -->
-    <link rel="preload" as="style" href="./dist/uniform.css">
```
[the remark this answers is not in the document]
Comment (alshakero) [reply index.html]: It turns out adding a preload tag does not add any benefit. I failed to find the issue where both Marcin and I came to this conclusion.

Do you remember where was this @warpech?

**Comment by alshakero** [comment]
All addressed.

File: index.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (tomalec) [reply index.html]: I also remember this issue, but the problem there was to add `preload` given we already have `<link rel="stylesheet">` in the same scope.

https://github.com/Starcounter/StarcounterClientFiles/pull/103#issuecomment-408487539


But in this case we do not have uniform in light dom, do we?

[the remark this answers is not in the document]
Comment (warpech) [reply index.html]: @tomalec is right, `preload` is not needed if `link rel="stylesheet"` is in light DOM. In this file, it is not in light DOM.

But then again, this file does not have to demonstrate preloading.

[the remark this answers is not in the document]
Comment (tomalec) [reply index.html]: > this file does not have to demonstrate preloading.

It does not mean it should not use preloading to improve UX.

**Pull request merged**